src/Network/GRPC/Common/HTTP2Settings.hs

```diff
@@ -0,0 +1,95 @@
+-- | Settings and parameters pertaining to HTTP\/2
+--
+-- Intended for unqualified import.
+
+module Network.GRPC.Common.HTTP2Settings
+  ( HTTP2Settings(..)
+  , defaultHTTP2Settings
+  ) where
+
+import Data.Default
+import Data.Word
+
+-- | HTTP\/2 settings
+data HTTP2Settings = HTTP2Settings {
+      -- | Maximum number of concurrent active streams
+      --
+      -- <https://datatracker.ietf.org/doc/html/rfc7540#section-5.1.2>
+      http2MaxConcurrentStreams :: Word32
+
+      -- | Initial window size for streams
+      --
+      -- <https://datatracker.ietf.org/doc/html/rfc7540#section-6.9.2>
+    , http2InitialStreamWindowSize :: Word32
```

Comment:
I wonder if we should call this `http2StreamWindowSize`; I find the _initial_ part a bit confusing. While it is true that the spec calls it this, and `http2` echoes the terminology, in the specific way that `http2` _uses_ this value it's not so much the _initial_ window as the _maximum_ window. In principle of course we could start at `X`, then at some point it decreases to say `X/2`, and then we send a window update for some `Y > X` -- but `http2` will never do that: it will only ever send window updates up to `X` again.

Reply:
I will remove the `initial` prefixes on these